File: N-Repeated_Element_In_Size_2N_Array/py.d/n-repeated.py
class Solution:
  def repeatedNTimes(self, A):
    counts = {}
    for i in A:
      if i not in counts:
        counts[i] = 1
      else:
        counts[i] += 1
      if counts[i] == len(A) / 2:
        return i

if __name__=="__main__":
  sol = Solution()
  print(Solution.repeatedNTimes([1,2,3,3])) # => 3

File: N-Repeated_Element_In_Size_2N_Array/js.d/n-repeated.js
/**
 * @param {number[]} A
 * @return {number}
 */
var repeatedNTimes = function(A) {
    counts = {};
    counts[A[0]] = 0;
    for (i=0; i<A.length; i++){
        //hasownproperty instead of loop;[]
        if (counts.hasOwnProperty(A[i])){
          counts[A[i]] += 1;
        } else {
          counts[A[i]] = 1;
        }
        if (counts[A[i]] == A.length / 2)
          return A[i];
    }
};

console.log(repeatedNTimes([1,2,3,3])) // => 3
